service/src/main/kotlin/fi/espoo/evaka/invoicing/data/FeeDecisionQueries.kt

```diff
@@ -326,7 +326,10 @@ fun Database.Read.searchFeeDecisions(
         if (numberParamsRaw.isNotEmpty()) numberQuery else null,
         if (searchTextWithoutNumbers.isNotBlank()) freeTextQuery else null,
         if ((startDate != null || endDate != null) && !searchByStartDate) "daterange(:start_date, :end_date, '[]') && valid_during" else null,
-        if ((startDate != null || endDate != null) && searchByStartDate) "daterange(:start_date, :end_date, '[]') @> lower(valid_during)" else null,
+        if ((startDate != null || endDate != null) && searchByStartDate)
+        // valid_during overlaps but does not extend to the left of search range
+        // = start date of valid_during is included in the search range
+            "(valid_during && daterange(:start_date, :end_date, '[]') valid_during &> daterange(:start_date, :end_date, '[]'))" else null,
```

Comment:
This is missing an `AND`.

Reply:
:facepalm: 

(also looks like we have 0 tests for this filter)